src/index.ts

```diff
@@ -323,6 +330,24 @@ export function offlineInit(config: IClientConfigSync): EppoClient {
  */
 export async function init(config: IClientConfig): Promise<EppoClient> {
   validation.validateNotBlank(config.apiKey, 'API key required');
+
+  // If there is already an init in progress for this apiKey, return that.
+  let _initPromise = _initializationPromises.get(config.apiKey);
+  if (_initPromise) {
+    return _initPromise;
+  }
+
+  _initPromise = explicitInit(config);
+  _initializationPromises.set(config.apiKey, _initPromise);
+
+  return _initPromise.then((client) => {
+    _initializationPromises.delete(config.apiKey);
+    return client;
+  });
```

Comment:
since this is an `async function`, you can just `await _initPromise` instead of using `then`, that's a bit cleaner

Reply:
agreed